### ISSUE TITLE
Notice: Trying to get property of non-object in /home/x/public_html/libraries/joomla/updater/adapters/collection.php on line 131

### ISSUE THREAD

**Issue by roland-d** [joomla/joomla-platform#1218]
Hello,

Sometimes when U run the CLI job update_cron.php I see the following 2 notice lines in the output:

Fetching updates...

Notice: Trying to get property of non-object in /home/x/public_html/libraries/joomla/updater/adapters/collection.php on line 131

Notice: Trying to get property of non-object in /home/x/public_html/libraries/joomla/updater/adapters/collection.php on line 131
Finished fetching updates

It doesn't always happen, so it is not constantly reproducible.

**Comment by eddieajau** [comment]
What's update_cron.php?

**Comment by roland-d** [comment]
It's a script that comes with Joomla and can be found in the cli folder.

**Comment by eddieajau** [comment]
Have you reported the issue on the CMS bug tracker?

**Comment by roland-d** [comment]
No because on the CMS tracker it says to report problems here with files in the libraries folder.

**Comment by eddieajau** [comment]
Well, I think you are going to have to report it to the CMS because
they aren't using the current platform.  Line 131 is a blank line.
There's a remote possibility that the bug has already been fixed in
the platform as well.

**Comment by elinw** [comment]
Obviously you can just go look and see where the former line 131 is located. 

https://github.com/joomla/joomla-cms/blob/master/libraries/joomla/updater/adapters/collection.php#L131

which has now moved to line 130 instead of 131
https://github.com/joomla/joomla-platform/blob/staging/libraries/joomla/updater/adapters/collection.php#L130

As far as I know from reading what Sam has said, client should always be referred to by name not a number. So that may be the issue.

**Comment by elinw** [comment]
I don't know if it would break things but this section of code is really mixing up cms stuff and platform stuff.  it's using JApplicationHelper::getClientInfo ... which is legacy and also the doc blocks for the getClient info method state that

```
/**
 * Gets information on a specific client id.  This method will be useful in
 * future versions when we start mapping applications in the database.
 *
 * This method will return a client information array if called
 * with no arguments which can be used to add custom application information.
```

So i would say that probably that what is now line 130 should probably be conditioned on the existence of $client->id. $attrs['CLIENT_ID'] = $client->id; 

I assume the platform has  a newer way to identify things

**Comment by elinw** [comment]
This can be closed as the fix has been merged.

**Comment by AmyStephen** [comment]
If $client->id is not set coming out of JApplication getClientInfo, there is a problem. Using an "isset" only prevents the error message from display, but it doesn't address why Joomla is unable to determine what Client it is running. 

It's probably more of a developer issue in that the application was not correctly initiated.  

https://github.com/joomla/joomla-platform/blob/staging/libraries/legacy/application/helper.php#L73getClientInfo

Not so sure it's a good idea to add this code since it seems to suggest it's acceptable to not be able to identify the client.

**Comment by dongilbert** [comment]
Closed. Fixed by #1304
